File: backend/413Final.API/Data/Entertainer.cs
using System.ComponentModel.DataAnnotations;
using Microsoft.AspNetCore.Mvc;

namespace _413Final.API.Data;

public class Entertainer
{
    [Key]
    [Required]
    public int EntertainerId { get; set; }
    public string EntStageName { get; set; }
    public string EntSSN { get; set; }
    public string EntStreetAddress { get; set; }
    public string EntCity { get; set; }
    public string EntState { get; set; }
    public string EntZipCode { get; set; }
    public string EntPhoneNumber { get; set; }
    public string? EntWebPage { get; set; }
    public string? EntEMailAddress { get; set; }
    public string DateEntered { get; set; }
    
    // Navigation property for one-to-many
    public List<Engagement> Engagements { get; set; }
}

File: frontend/types/entertainer.ts
export type Entertainer = {
  entertainerId: number;
  entStageName: string;
  entSSN: string;
  numberOfBookings: number;
  lastBookingDate: string | null;
  entStreetAddress: string;
  entCity: string;
  entState: string;
  entZipCode: string;
  entPhoneNumber: string;
  entWebPage?: string;
  entEMailAddress?: string;
  dateEntered: string;
};

// entertainer.ts (or wherever your types are defined)
export type EntertainerAddRequest = {
  entertainerId: number;
  entStageName: string;
  entSSN: string;
  entStreetAddress: string;
  entCity: string;
  entState: string;
  entZipCode: string;
  entPhoneNumber: string;
  entWebPage?: string;
  entEMailAddress?: string;
  dateEntered: string;
};
